stdlib/src/zlib.rs

```diff
@@ -587,4 +588,42 @@ mod zlib {
             Ok(Self::new(int))
         }
     }
+
+    #[pyattr]
+    #[pyclass(name = "_ZlibDecompressor")]
+    #[derive(Debug, PyPayload)]
+    pub struct ZlibDecompressor {
+        decompress: PyMutex<Decompress>,
+    }
+
+    impl Constructor for ZlibDecompressor {
+        type Args = ();
+
+        fn py_new(cls: PyTypeRef, _args: Self::Args, vm: &VirtualMachine) -> PyResult {
+            let decompress = Decompress::new(true);
+            let zlib_decompressor = ZlibDecompressor {
+                decompress: PyMutex::new(decompress),
+            };
+            zlib_decompressor
+                .into_ref_with_type(vm, cls)
+                .map(Into::into)
+        }
+    }
+
+    #[pyclass]
+    impl ZlibDecompressor {
+        #[pymethod]
+        fn decompress(&self, data: ArgBytesLike, vm: &VirtualMachine) -> PyResult<Vec<u8>> {
```

Comment:
By this definition, `decompress` always requires an ArgBytesLike argument.
When the argument doesn't fit, it raises TypeError. This article has more explanations how it works:
https://github.com/RustPython/RustPython/wiki/Python-Attributes

Reply:
Specifically
```
FAIL: test_Constructor (test.test_zlib.ZlibDecompressorTest.test_Constructor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/runner/work/RustPython/RustPython/pylib/Lib/test/test_zlib.py", line 971, in test_Constructor
    self.assertRaises(TypeError, zlib._ZlibDecompressor, "ASDA")
AssertionError: TypeError not raised by _ZlibDecompressor
```

Reply:
Even though
```rust
impl Constructor for ZlibDecompressor {
        type Args = ();

        fn py_new(cls: PyTypeRef, _args: Self::Args, vm: &VirtualMachine) -> PyResult {
            let decompress = Decompress::new(true);
            let zlib_decompressor = ZlibDecompressor {
                decompress: PyMutex::new(decompress),
            };
            zlib_decompressor
                .into_ref_with_type(vm, cls)
                .map(Into::into)
        }
    }
 ```

Reply:
In CPython,
```
>>> import zlib
>>> zlib._ZlibDecompressor("ASDA")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'str' object cannot be interpreted as an integer
```
_ZlibDecompressor has to reject str as first argument. ArgBytesLike accepts str.

You seem to miss `wbits` argument, which must be the first argument of `_ZlibDecompressor`

Reply:
ah ok that explains a lot, thanks

Reply:
However, `type Args = ()`, shouldn't that mean it rejects everything?

Reply:
I added a commit to fix this problem and the payload error. Constructor was not added to impl `_ZlibDecompressor`.

Reply:
Thank you so much, that fixed everything.